src/views/List.jsx

```diff
@@ -1,16 +1,48 @@
+import { useState } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	const [searchItem, setSearchItem] = useState('');
+
+	const handleChange = (e) => {
+		setSearchItem(e.target.value);
+	};
+
+	const filteredItems = data.filter((item) => {
+		return item.name.toLowerCase().includes(searchItem.toLowerCase());
+	});
+
 	return (
 		<>
-			<p>
-				Hello from the <code>/list</code> page!
-			</p>
+			<label htmlFor="item-search"> Search for an item: </label>
+			<input
+				id="item-search"
+				type="text"
+				placeholder="Search item..."
+				onChange={handleChange}
+				value={searchItem}
+			/>
+			{searchItem && (
+				<button
+					type="button"
+					name="clearInput"
```

Comment:
It's fine to leave this bit as is, but we don't need the `name` attribute here, since it's a button (I'm sorry if I included it in my previous comment).